src/navigation/navigation.cc

```diff
@@ -906,58 +790,48 @@ void Navigation::Run() {
     return;
   }
 
-  // Publish Navigation Status
-  if (nav_complete_) {
-    if (kDebug) printf("Nav complete\n");
-    Halt();
-    status_msg_.status = 3;
-    status_pub_.publish(status_msg_);
-    viz_pub_.publish(local_viz_msg_);
-    return;
-  }
-  status_msg_.status = 1;
-  status_pub_.publish(status_msg_);
+  ForwardPredict(time + params_.system_latency);
+
+  // Replan as necessary (Global Plan)
   if (!PlanStillValid()) {
-    Plan(nav_goal_loc_);
+    plan_path_ = Plan(robot_loc_, nav_goal_loc_);
   }
-  // Get Carrot.
+
+  // Get Carrot and check if done
   const Vector2f carrot = GetCarrot();
-  carrot_pub_.publish(CarrotToNavMsgsPath(carrot));
-  auto msg_copy = global_viz_msg_;
-  visualization::DrawCross(carrot, 0.2, 0x10E000, msg_copy);
-  visualization::DrawArc(
-      Vector2f(0, 0), params_.carrot_dist, -M_PI, M_PI, 0xE0E0E0, local_viz_msg_);
-  viz_pub_.publish(msg_copy);
   // Check if complete.
-  nav_complete_ = 
+  nav_complete_ =
       (robot_loc_ - carrot).squaredNorm() < Sq(params_.target_dist_tolerance) &&
       (robot_vel_).squaredNorm() < Sq(params_.target_dist_tolerance);
-  // Run local planner.
-  if (nav_complete_) {
-    if (kDebug) printf("Now complete\n");
-    Halt();
+  // Halt if necessary
+  if (nav_complete_ || abort_) {
```

Comment:
Currently not doing this because this abort_ variable is used to control the Halt behavior, during which the navigation is not complete and we probably want to resume. If you want, we can rename the abort_ variable.

Reply:
Ah, I see. Then yes, `abort_` should probably be `pause_`.